packages/components/src/Autocomplete/Menu.tsx

```diff
@@ -66,22 +94,32 @@ export function Menu({
   function setupKeyListeners() {
     useOnKeyDown("ArrowDown", (event: KeyboardEvent) => {
       if (!visible) return;
-
       event.preventDefault();
+
+      const requestedIndex = options[highlightedIndex + IndexChange.Next];
```

Comment:
Can we collapse the common functionality between `ArrowDown` and `ArrowUp` into a single method that these can then call. there is a decent amount of duplication between the two.